[DO NOT MERGE] App support features build PR

kollivier opened the pull request:
<!--
 1. Following guidance below, replace …'s with your own words
 2. After saving the PR, tick of completed checklist items
 3. Skip checklist items that are not applicable or not necessary
 4. Delete instruction/comment blocks
-->

### Summary
<!--
 * description of the change
 * manual verification steps performed
 * screenshots if the PR affects the UI
-->

This PR is used to generate builds containing in-progress features to be used by the Kolibri apps, particularly the Android version. This builds on top of the 0.13 branch of Kolibri.

Once the work on app-specific changes is completed, we will update this PR for merging into the 0.13 branch.

### Reviewer guidance
<!--
 * how can a reviewer test these changes?
 * are there any risky areas that deserve extra testing
-->

To test these builds, grab the whl file and follow the app's instructions on how to update the Kolibri installation to the whl contents.

### References
<!--
 * references to related issues and PRs
 * links to mockups or specs for new features
 * links to the diffs for any dependency updates, e.g. in iceqube or the perseus plugin
-->

…

----

### Contributor Checklist


PR process:

- [ ] PR has the correct target branch and milestone
- [ ] PR has 'needs review' or 'work-in-progress' label
- [ ] If PR is ready for review, a reviewer has been added. (Don't use 'Assignees')
- [ ] If this is an important user-facing change, PR or related issue has a 'changelog' label
- [ ] If this includes an internal dependency change, a link to the diff is provided

Testing:

- [ ] Contributor has fully tested the PR manually
- [ ] If there are any front-end changes, before/after screenshots are included
- [ ] Critical user journeys are covered by Gherkin stories
- [ ] Critical and brittle code paths are covered by unit tests

### Reviewer Checklist

- Automated test coverage is satisfactory
- PR is fully functional
- PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- External dependency files were updated if necessary (`yarn` and `pip`)
- Documentation is updated
- Contributor is in AUTHORS.md

julianduque committed:
Fix edit modals to navigate properly on save

- UserEditPage
- UserPermissionsPage
- ProfileEditPage

julianduque committed:
remove SUCCESS state on User Edit Page

jamalex committed:
Fixes to Android integrations to prevent errors

kollivier committed:
Merge pull request #6758 from jamalex/android_fixes

Fixes to Android integrations to prevent errors

jamalex committed:
Simplify the setup wizard when accessing it from an embedded webview

jamalex committed:
Remove direct dependency on ifcfg and use zeroconf.get_all_addresses

kollivier committed:
Merge pull request #6764 from jamalex/remove_ifcfg

Remove direct dependency on ifcfg and use zeroconf.get_all_addresses

jamalex committed:
Merge pull request #6763 from jamalex/simplified_onboarding

Simplify the setup wizard when accessing it from an embedded webview

jamalex committed:
Merge branch 'release-v0.13.x' of github.com:learningequality/kolibri into app-support

jamalex committed:
Add prototype "Share" button for Android intent-based sharing

rtibbles committed:
Add symlink capability check to system utils.

rtibbles committed:
Only run collectstatic once, do it conditionally on any upgrades happening.
Run with clear True and use symlinks when possible.

rtibbles committed:
Disable BasicAuth by default in production.

jamalex committed:
Merge branch 'release-v0.13.x' of github.com:learningequality/kolibri into app-support

jamalex committed:
Add parent topic name to Share text

radinamatic committed:
Adding support for Chinese Simplified

rtibbles committed:
Merge pull request #6805 from rtibbles/moving_forward_standing_still

Optimize static file loading

nucleogenesis committed:
hex the uuid to remove hyphens

nucleogenesis committed:
no need to update unless anonymous

nucleogenesis committed:
Merge pull request #6827 from nucleogenesis/hex-the-uuid

hex the uuid to remove hyphens

lyw07 committed:
fix test_extract_channel_statistics when there are no users with a specific gender

indirectlylit committed:
Merge pull request #6838 from lyw07/fix-channel-stats-test

fix test_extract_channel_statistics test

radinamatic committed:
Missing comma

lyw07 committed:
update piwheels url for cryptography rpi packages download

lyw07 committed:
Merge pull request #6858 from lyw07/fix-piwheels-url-0.12

update piwheels url for cryptography rpi packages download - 0.12

lyw07 committed:
update piwheels url for cryptography rpi packages download

jonboiser committed:
Merge pull request #6860 from lyw07/fix-piwheels-url-0.13

rtibbles committed:
Initial implementation of app plugin.

indirectlylit committed:
Merge pull request #6820 from radinamatic/test-ZH-only

Adding support for Chinese Simplified

jamalex committed:
CONTENT_FALLBACK_DIRS option along with cherrypy/devserver support.

jamalex committed:
Get scanforcontent/upgrade working with fallback content dirs

jamalex committed:
Add scanforcontent args to filter by channel and modify import behavior

rtibbles committed:
Vendor diskcache RLock to use django cache transparently.
Don't recreate diskcache at import.
Only recreate diskcache when completely necessary.

nucleogenesis committed:
Merge remote-tracking branch 'upstream/release-v0.12.x' into merge-12-to-13-but-do-it-right-this-time

indirectlylit committed:
updates to release process docs

radinamatic committed:
Text formatting and minor edits

jonboiser committed:
Merge pull request #6859 from nucleogenesis/merge-0.12

indirectlylit committed:
Merge pull request #31 from radinamatic/chill-freeze

Text formatting and minor edits

jredrejo committed:
Apply restrictions only to learners

jredrejo committed:
fixed jest test

jredrejo committed:
Update gherkin scenario according to these changes

jredrejo committed:
Catch error when kolibri runs in linux non-debian distros

rtibbles committed:
Merge pull request #6847 from learningequality/release-v0.13.x

Release v0.13.x -> app-support branch

jredrejo committed:
Added python 2.0 compatibility

indirectlylit committed:
Merge remote-tracking branch 'le/release-v0.13.x' into release-process-docs

indirectlylit committed:
updates based on Lingy's feedback

indirectlylit committed:
Merge pull request #6883 from jredrejo/non_debian_systems

Catch error when kolibri runs in linux non-debian distros

DXCanas committed:
Reintroducing android to Kolibri pipeline

indirectlylit committed:
Merge pull request #6877 from indirectlylit/release-process-docs

Release process docs update

jamalex committed:
Validate the path list for KOLIBRI_CONTENT_FALLBACK_DIRS

jamalex committed:
Add a short-circuit if there's only one possible handler to dispatch

jamalex commented:
Build failure looks to be a result of the same thing as https://github.com/learningequality/kolibri/pull/6835#issuecomment-624350987

jamalex committed:
Merge branch 'app-support' of github.com:learningequality/kolibri into content_fallback_paths

jamalex committed:
Fix path expansion logic for multi-path lists

jredrejo committed:
Adjust Spanish translations

jamalex committed:
Fix get_content_fallback_pathspaths function to handle list, not str

indirectlylit committed:
arrow buttons on very small screens

jredrejo committed:
Facility settings changes for passwords

jamalex committed:
run black on updated code

jamalex committed:
Bump zeroconf to 0.19.6 for flatpak-compatibility

indirectlylit committed:
Merge pull request #6914 from indirectlylit/quiz-buttons

arrow buttons on very small screens

jredrejo committed:
When self-creating an account, password is not requested

jredrejo committed:
Fix current js test

jredrejo committed:
lint fix

jonboiser committed:
Remove in-page progress messages in UserPermissionsPage

since they’re not shown anymore

rtibbles committed:
Simplify process cache import.

jredrejo committed:
Added tooltip and hide password to the usercreation

lyw07 committed:
add option STATIC_USE_SYMLINKS to toggle on/off symlinked static files

jamalex committed:
Merge pull request #4 from learningequality/app-support

App support

jamalex committed:
Merge pull request #6865 from jamalex/content_fallback_paths

CONTENT_FALLBACK_DIRS option along with cherrypy/devserver support.

rtibbles committed:
Add process cache to test settings to properly test RLock behaviour.

lyw07 committed:
bump perseus plugin version to 1.3.2

rtibbles committed:
Add device app key and endpoint for validating the app key.

rtibbles committed:
Add AppView permission to AppCommands.

rtibbles committed:
Check appview cookie in JS appview check.

rtibbles committed:
Make capability reading robust for tests.

rtibbles committed:
Lint migration.

jredrejo committed:
Remove tooltip

rtibbles committed:
Fix h5p loading for Python3.
Add more semantic errors.

rtibbles committed:
Merge pull request #6927 from lyw07/bump-perseus-to-1.3.2

bump perseus plugin version to 1.3.2

lyw07 committed:
Merge pull request #6895 from DXCanas/build-test-android-reintro

Reintroduction of Android Builds

rtibbles committed:
Fix tests for functionality dependent on process_cache behaviour.

rtibbles committed:
Use process cache in dev to give a more veridical dev experience.

jredrejo committed:
unify the way non_specified passwords are created

lyw07 committed:
add a docstring to notions.py module to explain option STATIC_USE_SYMLINKS

jredrejo committed:
some lint and prettier fixes

jredrejo committed:
Fix more linting errors

rtibbles committed:
Properly catch error.

indirectlylit committed:
Merge pull request #6925 from lyw07/toggle-off-symlinked-static

add option STATIC_USE_SYMLINKS to toggle on/off symlinked static files

indirectlylit committed:
Merge pull request #6924 from jonboiser/coach-double-save

Adds exit-on-success behavior from #6705

jonboiser committed:
Merge pull request #6870 from rtibbles/redis_cache_process_cache

rtibbles committed:
Fix json.load typo

rtibbles committed:
Merge pull request #6933 from indirectlylit/basic-auth

Disable BasicAuth by default in production.

jonboiser committed:
Merge pull request #6928 from rtibbles/json_from_zip

rtibbles committed:
Remove first_login checking.
Use lack of installed content to prompt welcome modal.
Always redirect super users to Device.

radinamatic committed:
Change the name for Simplified Chinese in the language modal

radinamatic committed:
Change the name in base.py too

rtibbles committed:
Address PR comments.

lyw07 committed:
Merge pull request #6939 from radinamatic/zh-name

Change the name for Simplified Chinese in the language modal

indirectlylit committed:
Merge pull request #6931 from learningequality/release-v0.13.x

Release v0.13.x -> app support

rtibbles committed:
Update tests for new redirect behaviour.

indirectlylit committed:
Merge pull request #6923 from jredrejo/remote_access_settings

Facility settings updates for app suport

indirectlylit committed:
latest minor i18n updates

Commit:
Some fixes on the SignUpPage.vue

rtibbles committed:
Update based on further feedback.

rtibbles committed:
Block welcome modal display once dismissed within a session.

rtibbles committed:
Coerce keys when generating channel_stats rather than when accessing it.

indirectlylit committed:
Merge pull request #6951 from jredrejo/more_facility_settings

Some facility settings fixes

rtibbles committed:
Update naming of permission class.

indirectlylit committed:
fix for manifest, ref #6796 (issue 3)

jonboiser committed:
Revert renaming 'zh-Hans.js' path to 'zh-HANS.js'

See Issue #6796, for discussion on i18n code generation tools

indirectlylit committed:
update to capitalization code, ref #6796

indirectlylit committed:
update to documentation, ref #6796

rtibbles committed:
Merge pull request #6864 from rtibbles/app_plugin

Initial implementation of app plugin.

indirectlylit committed:
update capitalization

indirectlylit committed:
Merge branch 'i18n-013' of github.com:indirectlylit/kolibri into i18n-013

rtibbles committed:
Merge pull request #6938 from rtibbles/dismiss_first_login

Prevent the overly persistent welcome modal

jonboiser committed:
Merge pull request #6949 from indirectlylit/i18n-013

rtibbles committed:
Merge pull request #6956 from rtibbles/uuids_importability

Convert keys from UUIDs to strings when channel stats are created, not accessed.

indirectlylit committed:
Merge remote-tracking branch 'le/release-v0.13.x' into app-support

indirectlylit committed:
Merge pull request #6961 from indirectlylit/app-support

0.13 -> App support
